### PR TITLE
*: replace jwt mod with `golang-jwt`

### LINKED ISSUE #37091: we should update `jwt` dependency because of `github.com/form3tech-oss/jwt-go` was archived
## General Question

`github.com/form3tech-oss/jwt-go` is archived, and they recommend to migrate to [`github.com/golang-jwt/jwt`](https://github.com/golang-jwt/jwt)

How about we migrate jwt to `github.com/golang-jwt/jwt`?

### PR DESCRIPTION
**Pull request by wuhuizuo**
### What problem does this PR solve?

Issue Number: close #37091 

Problem Summary:

### What is changed and how it works?

replace archived jwt mod with `glang-jwt`

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [x] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -451,7 +451,7 @@ bazel_test: failpoint-enable bazel_prepare
 
 
 bazel_coverage_test: check-bazel-prepare failpoint-enable bazel_ci_prepare
-	bazel $(BAZEL_GLOBAL_CONFIG) --nohome_rc coverage $(BAZEL_CMD_CONFIG) --jobs=35 --build_tests_only --test_keep_going=false \
+	bazel $(BAZEL_GLOBAL_CONFIG) coverage $(BAZEL_CMD_CONFIG) --build_tests_only --test_keep_going=false \
```
Comment (wuhuizuo) [inline Makefile:454]:
```suggestion
	bazel $(BAZEL_GLOBAL_CONFIG) --nohome_rc coverage $(BAZEL_CMD_CONFIG) --jobs=35 --build_tests_only --test_keep_going=false \
```

File: executor/BUILD.bazel [hunk 2]
```diff
@@ -301,6 +301,7 @@ go_test(
         "explain_test.go",
         "explain_unit_test.go",
         "explainfor_test.go",
+        "foreign_key_test.go",
```
Comment (wuhuizuo) [inline executor/BUILD.bazel:304]:
```suggestion
        "foreign_key_test.go",
```

**Comment by wuhuizuo** [comment]
Ref: https://github.com/pingcap/tidb/issues/46207

What the error it happened?

> @wuhuizuo: The following tests **failed**, say `/retest` to rerun all failed tests or `/retest-required` to rerun all mandatory failed tests:
> 
> Test name	Commit	Details	Required	Rerun command
> canary-unit-test	[521ab0d](https://github.com/pingcap/tidb/commit/521ab0d31c889434f903e9464942db5a21aeac07)	[link](https://do.pingcap.net/jenkins/job/pingcap/job/tidb/job/canary_ghpr_unit_test/3/display/redirect)	true	`/test pingcap/tidb/canary_ghpr_unit_test`
> pull-br-integration-test	[521ab0d](https://github.com/pingcap/tidb/commit/521ab0d31c889434f903e9464942db5a21aeac07)	[link](https://do.pingcap.net/jenkins/job/pingcap/job/tidb/job/pull_br_integration_test/136/display/redirect)	true	`/test pull-br-integration-test`
> idc-jenkins-ci-tidb/check_dev	[3ba61d1](https://github.com/pingcap/tidb/commit/3ba61d1cd889c702abbee6cf367fcbfd662eb745)	[link](https://do.pingcap.net/jenkins/job/pingcap/job/tidb/job/ghpr_check/26770/display/redirect)	true	`/test check-dev`
> idc-jenkins-ci-tidb/unit-test	[3ba61d1](https://github.com/pingcap/tidb/commit/3ba61d1cd889c702abbee6cf367fcbfd662eb745)	[link](https://do.pingcap.net/jenkins/job/pingcap/job/tidb/job/ghpr_unit_test/31486/display/redirect)	true	`/test unit-test`
> [Full PR test history](https://prow.tidb.net/pr-history?org=pingcap&repo=tidb&pr=37240). [Your PR dashboard](https://prow.tidb.net/pr?query=is%3Apr%20state%3Aopen%20author%3Awuhuizuo).

**Comment by Defined2014** [comment]
Not repy it anymore, we could remove it direclty. So close this PR.

**Pull request closed without merging**